src/tools/clippy/clippy_lints/src/operators/assign_op_pattern.rs

```diff
@@ -45,10 +44,8 @@ pub(super) fn check<'tcx>(
                 }
 
                 // Skip if the trait is not stable in const contexts
-                if is_in_const_context(cx)
-                    && let Some(binop_id) = cx.tcx.associated_item_def_ids(trait_id).first()
-                    && !is_stable_const_fn(cx, *binop_id, msrv)
-                {
+                // FIXME: reintroduce a better check after this is merged back into Clippy
+                if is_in_const_context(cx) {
```

Comment:
Wouldn't we want to check for `tcx.features().const_ops()`? (I'm not being able to find `const_ops` on `Features`)

Reply:
Have you seen the discussion starting at https://github.com/rust-lang/rust/pull/143949#discussion_r2208466376?

Reply:
Note, the proper fix should be done on the clippy repo later. The issue is that the bug is entirely unrelated to this change, but the changes make the existing test no longer pass, so, something has to give, and this was the minimal fix to at least get rid of some invalid suggestions.

Reply:
`const_ops` is a library feature so `tcx.features().const_ops()` doesn't exist. `tcx.features().enabled(...)` should work though.

Cc @flip1995

Reply:
FWIW, there shouldn't be any feature checking here since both the assignment version of the operators and the non-assignment version are gated behind the same features, so, if you can use the non-assignment version, you will be able to use the assignment version.

The issue is that this isn't checking specifically for the bounds on both operators, since `[const] Add` doesn't guarantee `[const] AddAssign`. It does check that `AddAssign` is a bound when suggesting it, but it isn't also checking for `[const] AddAssign` when you're calling it in a const context.